### ISSUE TITLE
[Keyboard a11y] Delete popover in playlist index page

### ISSUE THREAD

**Issue by Dananji** [avalonmediasystem/avalon#6355]
### Description

From accessibility checks performed in https://github.com/avalonmediasystem/avalon/issues/6312, it was identified that the buttons within the delete popover in the playlist index page cannot be reached with keyboard.

Steps to reproduce:
1. Use `Tab` navigation to reach the `Delete` button for any playlist in the table display
2. Press `Enter`
3. Observe the delete popover being opened
4. Try to reach the `Delete` or `Cancel` button in the popover using keyboard

### Done Looks Like
- [x] A keyboard user can reach and use the `Delete`/`Cancel` buttons within the delete popover

**Comment by Dananji** [comment]
From [Bootstrap documentation](https://getbootstrap.com/docs/4.6/components/popovers/);

> ### Making popovers work for keyboard and assistive technology users

> To allow keyboard users to activate your popovers, you should only add them to HTML elements that are traditionally keyboard-focusable and interactive (such as links or form controls). Although arbitrary HTML elements (such as <span>s) can be made focus-able by adding the `tabindex="0"` attribute, this will add potentially annoying and confusing tab stops on non-interactive elements for keyboard users, and most assistive technologies currently do not announce the popover’s content in this situation. Additionally, do not rely solely on hover as the trigger for your popovers, as this will make them impossible to trigger for keyboard users.

> While you can insert rich, structured HTML in popovers with the `html` option, we strongly recommend that you avoid adding an excessive amount of content. The way popovers currently work is that, once displayed, their content is tied to the trigger element with the `aria-describedby` attribute. As a result, the entirety of the popover’s content will be announced to assistive technology users as one long, uninterrupted stream.

> Additionally, while it is possible to also include interactive controls (such as form elements or links) in your popover (by adding these elements to the `whiteList` or allowed attributes and tags), be aware that currently the popover does not manage keyboard focus order. When a keyboard user opens a popover, focus remains on the triggering element, and as the popover usually does not immediately follow the trigger in the document’s structure, there is no guarantee that moving forward/pressing TAB will move a keyboard user into the popover itself. In short, simply adding interactive controls to a popover is likely to make these controls unreachable/unusable for keyboard users and users of assistive technologies, or at the very least make for an illogical overall focus order. In these cases, consider using a modal dialog instead

According to this, popovers are not an accessible-friendly solution to be used for this specific UI feature. Because the popover we are displaying in this context contains actionable buttons in it, which a keyboard user cannot reach with the way popovers are designed.

However, [Mozilla developer documentation](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using#popover_accessibility_features) suggests that Popover API can be used to make such UI elements keyboard accessible;

>When a relationship is established between a popover and its control (invoker) via the `popovertarget` attribute, the API automatically makes two other changes to the environment to allow keyboard and assistive technology (AT) users to more easily interact with the popover:
>- When the popover is shown, the keyboard focus navigation order is updated so that the popover is next in the sequence: for example, when a button is pressed to show a popover, any buttons inside the popover will be next in the tabbing order (will be focused by pressing the Tab key). Conversely, when closing the popover via the keyboard (usually via the Esc key), focus is shifted back to the invoker.
>- To allow AT such as screen readers to make sense of the relationship between the invoker and the popover, an implicit [`aria-details`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-details) and [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-expanded) relationship is set up between them.

I am going to try the second implementation to see whether we can make it work.

**Comment by Dananji** [comment]
I ended up using events related to popovers in Bootstrap since we are using Bootstrap popovers in the code instead of the Popover API. We are leveraging Bootstrap popover events in JS to move and trap focus as needed when handling popovers via keyboard.
